.circleci/config.yml

```diff
@@ -695,46 +713,98 @@ jobs:
       - *restore-cache-gems
-      - *restore-cache-podmaster
       - *install-gems
       - *prepare-netrc-file
       - *add-github-to-known-hosts
-      - install-mbx-ci
       - run:
           name: Generate docs
           command: |
-            OUTPUT=$( echo << pipeline.git.tag >> | sed 's/^v//' )
-            export OUTPUT="${OUTPUT:-documentation}"
-            echo "export OUTPUT=$OUTPUT" >> $BASH_ENV
-            ./scripts/document.sh
-            cp -a "$OUTPUT" /tmp/generated-docs
-      - *save-cache-podmaster
+            TAG="<< pipeline.git.tag >>"
+            if [ -n "$TAG" ]; then
+              VERSION=${TAG#v}
+            else
+              VERSION=0.0.0
+            fi
+            OUTPUT_PATH="build/docs/$VERSION"
+            echo "export OUTPUT_PATH=$OUTPUT_PATH" >> $BASH_ENV
+            ./scripts/document.sh "$VERSION" "$OUTPUT_PATH"
       - *save-cache-gems
+      - persist_to_workspace:
+          root: ./
+          paths:
+            - build/docs
       - run:
-          name: Compress Docs
-          command: tar -cvzf /tmp/docs.tar /tmp/generated-docs
+          name: Compress docs
+          command: |
+            mkdir -p build/archives
+            tar -cvzf build/archives/navsdk-v2-docs.tar $OUTPUT_PATH
       - store_artifacts:
-          path: /tmp/docs.tar
-          destination: docs
+          path: build/archives/navsdk-v2-docs.tar
+          destination: navsdk-v2-docs
+      - notify-build-finished
+
+  publish-docs-job:
+    docker:
+      - image: cimg/base:current
+    resource_class: small
+    environment:
+      HOMEBREW_NO_AUTO_UPDATE: 1
+      MBX_CI_DOMAIN: o619qyc20d.execute-api.us-east-1.amazonaws.com
+      TARGET_ENVIRONMENT: docs.mapbox.com
+    steps:
       - when:
-          condition: << pipeline.git.tag >>
+          condition: << pipeline.git.tag >> =~ /^v2\.\d+\.\d+(-.+)?$/
           steps:
-            - run:
-                name: Push docs to publisher-production
-                command: |
-                  git remote set-url origin "https://x-access-token:$(mbx-ci github writer public token)@github.com/mapbox/mapbox-navigation-ios.git"
-                  git config user.email "release-bot@mapbox.com"
-                  git config user.name "Mapbox Releases"
-                  VERSION="${OUTPUT}" scripts/publish-docs.sh
-            - notify-build-finished:
-                notify_success: true
-      - notify-build-finished
+          - checkout
+          - *add-github-to-known-hosts
+          - attach_workspace:
+              at: .
+          - install-linux-mbx-ci
+          - install-linux-s5cmd
+          - run:
+              name: Publish docs
+              command: |
+                TAG="<< pipeline.git.tag >>"
+                if [ -n "$TAG" ]; then
+                  VERSION=${TAG#v}
+                else
+                  echo "ERROR: No tag found"
+                  exit 1
+                fi
+
+                SOURCE_PATH="build/docs/$VERSION"
```

Comment:
Why not store this script in a separate file, `publish-docs.sh`? This step is quite big, so it just makes sense to keep it in a separate file instead of removing the existing script file